rust_snuba/rust_arroyo/src/processing/strategies/run_task_in_threads.rs

```diff
@@ -127,6 +127,12 @@ impl<TPayload, TTransformed: Send + Sync + 'static> ProcessingStrategy<TPayload>
         self.metrics
             .gauge(&self.metric_name, self.handles.len() as u64, None);
 
+        self.metrics.gauge(
+            "arroyo.strategies.run_task_in_threads.concurrency",
```

Comment:
can you make it so that we use `strategy_name`, like we do for the gauge metric above?

our actual problem is, we have multiple uses of RunTaskInThreads: once for the message processing, and once for the clickhouse inserts

this means that we needed a way to distinguish between teh metrics of the various strategy instances.

I suggest adding a new field like `metric_name` (but better named) to `self` that is formatted a similar way.

```
self.metrics.gauge(self.concurrency_metrics_name, ...)
```

alternatively you can also refactor those metrics to use a metric tag with a value of `strategy_name`, instead of formatting the `strategy_name` into the metric tag:

```
self.metrics.gauge("arroyo.strategies.run_task_in_threads.concurrency", ..., Some(HashMap::from(["strategy_name", self.strategy_name]))
```

I would actually prefer the latter

Reply:
Not sure if this is the best way to do it,  but I switched it up to use a shared prefix that I append the metric name to.